src/network/networkinterfacemanager.cpp

```diff
@@ -24,71 +27,469 @@ namespace aos::common::network {
  * Public
  **********************************************************************************************************************/
 
-Error NetworkInterfaceManager::RemoveInterface(const String& ifname)
+Bridge::Bridge(const sm::networkmanager::LinkAttrs& attrs)
+    : mAttrs(attrs)
+{
+}
+
+const sm::networkmanager::LinkAttrs& Bridge::GetAttrs() const
+{
+    return mAttrs;
+}
+
+const char* Bridge::GetType() const
+{
+    return "bridge";
+}
+
+Vlan::Vlan(const sm::networkmanager::LinkAttrs& attrs, int vlanId)
+    : mAttrs(attrs)
+    , mVlanId(vlanId)
+{
+}
+
+const sm::networkmanager::LinkAttrs& Vlan::GetAttrs() const
+{
+    return mAttrs;
+}
+
+const char* Vlan::GetType() const
+{
+    return "vlan";
+}
+
+int Vlan::GetVlanId() const
+{
+    return mVlanId;
+}
+
+Error NetworkInterfaceManager::DeleteLink(const String& ifname)
 {
     LOG_DBG() << "Remove interface: ifname=" << ifname;
 
-    auto sock = socket(AF_NETLINK, SOCK_RAW, NETLINK_ROUTE);
-    if (sock < 0) {
-        return Error(errno, "failed to create netlink socket");
+    auto sockDeleter = [](nl_sock* sock) { nl_socket_free(sock); };
+    auto sock        = std::unique_ptr<nl_sock, decltype(sockDeleter)>(nl_socket_alloc(), sockDeleter);
+    if (!sock) {
+        return Error(
+            ErrorEnum::eFailed, ("failed to allocate netlink socket: " + std::string(nl_geterror(errno))).c_str());
```

Comment:
Consider to add static/private function: `NLToAosError'

Reply:
Done

Reply:
Where? I don't see these changes.

Reply:
Forgot to update